examples/weather/modules/weather-shared-data/src/main/scala/my/weathermetagraph/shared/data/domain/MetagraphState.scala

```diff
@@ -0,0 +1,140 @@
+package my.weathermetagraph.shared.data.domain
+
+import cats.effect.Sync
+import cats.syntax.functor._
+import derevo.circe.magnolia.{decoder, encoder}
+import derevo.derive
+import io.constellationnetwork.metagraph_sdk.std.JsonBinaryCodec
+import my.weathermetagraph.shared.data.domain.Account.AccountState
+import my.weathermetagraph.shared.data.domain.Location.LocationRecord
+import my.weathermetagraph.shared.data.domain.Request.RequestUpdate
+import my.weathermetagraph.shared.data.domain.Weather.WeatherCondition.WeatherCondition
+import my.weathermetagraph.shared.data.domain.Weather.WeatherRecord
+import org.tessellation.currency.dataApplication.{DataCalculatedState, DataOnChainState, DataUpdate}
+import org.tessellation.schema.SnapshotOrdinal
+
+import java.time.Instant
+
+object MetagraphState {
+
+  @derive(encoder, decoder)
+  sealed trait WeatherDataUpdate extends DataUpdate
+
+  @derive(encoder, decoder)
+  sealed trait WeatherDataCalculatedState {
+    def updates: List[WeatherDataUpdate]
+  }
+
+  @derive(encoder, decoder)
+  sealed trait WeatherDataOnChainState {
+    def updates: List[WeatherDataUpdate]
+  }
+
+  @derive(encoder, decoder)
+  case class WeatherUpdate(
```

Comment:
with these being updates, does this mean that any user can submit weather updates? How do you restrict who is allowed to push new values?